src/wp-includes/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -579,6 +579,12 @@ private function evaluate( $directive_value ) {
 		$path_segments = explode( '.', $path );
 		$current       = $store;
 		foreach ( $path_segments as $path_segment ) {
+			// Special case for numeric arrays to add a length property like JavaScript arrays.
+			if ( 'length' === $path_segment && is_array( $current ) && array_is_list( $current ) ) {
+				$current = count( $current );
+				break;
+			}
```

Comment:
This should work for strings as well.

That would seem to cover the instance properties available on JavaScript object that would be immediately accessible on the client via directives:

```
<div data-wp-hidden="!state.stringValue.length" data-wp-text="state.stringValue"></div>
```